features/situation-feedback/api/src/main/java/org/opennms/features/situationfeedback/api/AlarmFeedback.java

```diff
@@ -61,8 +65,12 @@ public static FeedbackType valueOfOrUnknown(String type) {
 
     private final FeedbackType feedbackType;
 
+    private final Boolean rootCause;
+
     private final String reason;
 
+    private final List<String> tags = new ArrayList<>();;
```

Comment:
Extra semicolon.

Also,

I suspect this should be a `Set` as I don't think a tag occurring twice would be meaningful. Not sure if ordering matters here, if so used a `LinkedHashSet`.

```suggestion
    private final List<String> tags = new ArrayList<>();
```

Reply:
In the json that this serialized to and from, it will be a List

Reply:
The serialization should only care about the setter in the builder, in this case `withTags`. So the argument that takes can remain a list, but the internal data structure we use can be a set. This will let us de-duplicate immediately upon receiving tags from the client and we won't end up storing dupes.

Reply:
Feedback also goes in and out the other end (ElasticSearch) as a JSON array. I'd rather keep the internal structure as a List to keep it in line with the external representations